python/cudf/dataframe/dataframe.py

```diff
@@ -2151,12 +2151,34 @@ def from_records(self, data, index=None, columns=None, nan_as_null=False):
         -------
         DataFrame
         """
-        names = data.dtype.names if columns is None else columns
+        if data.ndim != 1 and data.ndim != 2:
+            raise ValueError("records dimension expected 1 or 2 but found {!r}"
+                             .format(data.ndim))
+
+        num_cols = len(data[0])
+        if columns is None and data.dtype.names is None:
+            names = [i for i in range(num_cols)]
+
+        elif data.dtype.names is not None:
+            names = data.dtype.names
+
+        else:
+            if len(columns) != num_cols:
+                msg = "columns length expected {!r} but found {!r}"
+                raise ValueError(msg.format(num_cols, len(columns)))
+            names = columns
+
         df = DataFrame()
-        for k in names:
-            # FIXME: unnecessary copy
-            df[k] = Series(np.ascontiguousarray(data[k]),
-                           nan_as_null=nan_as_null)
+        if data.ndim == 2:
+            for i, k in enumerate(names):
+                # FIXME: unnecessary copy
+                df[k] = Series(np.ascontiguousarray(data[:, i]),
```

Comment:
We handle making this contiguous under the hood now so no need to do this here:
```suggestion
                df[k] = Series(data[:, i],
```

Reply:
sure